### PR TITLE
CLN: remove the circular import in NDFrame.dtypes

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5366,9 +5366,8 @@ def dtypes(self):
         string              object
         dtype: object
         """
-        from pandas import Series  # noqa: F811
-
-        return Series(self._mgr.get_dtypes(), index=self._info_axis, dtype=np.object_)
+        data = self._mgr.get_dtypes()
+        return self._constructor_sliced(data, index=self._info_axis, dtype=np.object_)
```
Comment (jbrockmendel) [inline pandas/core/generic.py:5370]: is this overriden by Series?

Comment (MarcoGorelli) [reply pandas/core/generic.py:5370]: It is in `frame.py` - https://github.com/pandas-dev/pandas/blob/19e8fcc2719575ff0d6a444d035645d009aa2bfa/pandas/core/frame.py#L420

And for `Series` the `dtypes` property is overridden

**Comment by WillAyd** [comment]
lgtm. doc build failure looks unrelated but restarted to be sure

**Review by WillAyd — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Pull request merged**